lib/RxTouch.js

```diff
@@ -1,8 +1,105 @@
 const s3 = require("../lib/s3");
 
-module.exports.noAppointmentMatchString = "no locations with available appointments";
+async function jQueryPost(page, url, data) {
+    return await page.evaluate(
+        async (url, data) => {
+            return await new Promise((resolve) => {
+                $.post(url, data, function (data) {
+                    resolve(data);
+                });
+            });
+        },
+        url,
+        data
+    );
+}
 
-module.exports.ScrapeRxTouch = async function ScrapeRxTouch(browser, site, siteName) {
+async function GetAllAvailability(availabilityService, zip) {
+    const today = new Date();
+    const currYear = today.getFullYear();
+    const monthNumsToCheck = [
+        // we'll need to fix this in Dec 2021... hopefully we don't need this project then
+        today.getMonth() + 1,
+        today.getMonth() + 2,
+    ];
+    const monthlyAvailability = await Promise.all(
+        monthNumsToCheck.map(async (monthNum) => {
+            const response = await availabilityService.getAvailabilityForMonth(
+                currYear,
+                monthNum
+            );
+            return {
+                monthNum,
+                availability: response?.Success
+                    ? response?.Data?.Days?.filter((day) => day.Available)
+                    : null,
+            };
+        })
+    );
+    const dailyAvailability = await Promise.all(
+        monthlyAvailability.map(async (entry) => {
+            return await Promise.all(
+                entry.availability.map(async (day) => {
+                    const response = await availabilityService.getAvailabilityForDay(
+                        currYear,
+                        entry.monthNum,
+                        day.DayNumber
+                    );
+                    return {
+                        date: `${entry.monthNum}/${day.DayNumber}/${currYear}`,
+                        numAppointments: response?.Success
+                            ? response?.Data?.Rows?.length
```

Comment:
we're relying on this to be a number... I suggest doing
`response?.Data?.Rows?.length || 0` to make sure that we get a number for that portion of the ternary.